File: app/src/main/java/com/example/lvchen/myapplication/MainActivity.java
package com.example.lvchen.myapplication;

import android.app.WallpaperManager;
import android.content.Context;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.drawable.BitmapDrawable;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.NavigationView;
import android.support.design.widget.Snackbar;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;

public class MainActivity extends AppCompatActivity
		implements NavigationView.OnNavigationItemSelectedListener, View.OnClickListener {

	private ImageView imageView;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
		setSupportActionBar(toolbar);

		FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
		fab.setOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
				Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
						.setAction("Action", null).show();
			}
		});

		DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
		ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
				this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
		drawer.setDrawerListener(toggle);
		toggle.syncState();

		NavigationView navigationView = (NavigationView) findViewById(R.id.nav_view);
		navigationView.setNavigationItemSelectedListener(this);

		Button button = (Button) findViewById(R.id.granzort_btn);
		button.setOnClickListener(this);

		Button viewpagerBtn = (Button) findViewById(R.id.viewpager_btn);
		viewpagerBtn.setOnClickListener(this);

		Button wallpaperBtn = (Button) findViewById(R.id.wallpaper_btn);
		wallpaperBtn.setOnClickListener(this);

		imageView = (ImageView) findViewById(R.id.imageView);
	}

	@Override
	public void onBackPressed() {
		DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
		if (drawer.isDrawerOpen(GravityCompat.START)) {
			drawer.closeDrawer(GravityCompat.START);
		} else {
			super.onBackPressed();
		}
	}

	@Override
	public boolean onCreateOptionsMenu(Menu menu) {
		// Inflate the menu; this adds items to the action bar if it is present.
		getMenuInflater().inflate(R.menu.main, menu);
		return true;
	}

	@Override
	public boolean onOptionsItemSelected(MenuItem item) {
		// Handle action bar item clicks here. The action bar will
		// automatically handle clicks on the Home/Up button, so long
		// as you specify a parent activity in AndroidManifest.xml.
		int id = item.getItemId();

		//noinspection SimplifiableIfStatement
		if (id == R.id.action_settings) {
			return true;
		}

		return super.onOptionsItemSelected(item);
	}

	@SuppressWarnings("StatementWithEmptyBody")
	@Override
	public boolean onNavigationItemSelected(MenuItem item) {
		// Handle navigation view item clicks here.
		int id = item.getItemId();

		if (id == R.id.nav_camera) {
			// Handle the camera action
		} else if (id == R.id.nav_gallery) {
			startActivity(new Intent(MainActivity.this, RecycleViewActivity.class));
		} else if (id == R.id.nav_slideshow) {

		} else if (id == R.id.nav_manage) {

		} else if (id == R.id.nav_share) {

		} else if (id == R.id.nav_send) {

		}

		DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
		drawer.closeDrawer(GravityCompat.START);
		return true;
	}

	@Override
	public void onClick(View view) {
		switch (view.getId()) {
			case R.id.granzort_btn:
				startActivity(new Intent(MainActivity.this, GranzortViewActivity.class));
				break;
			case R.id.viewpager_btn:
				startActivity(new Intent(MainActivity.this, ViewPagerActivity.class));
				break;
			case R.id.wallpaper_btn:
				WallpaperManager wallpaperManager = WallpaperManager
						.getInstance(getApplicationContext());
				// 获取当前壁纸
				BitmapDrawable drawable = (BitmapDrawable) wallpaperManager.getDrawable();
				Log.d("MainActivity", "drawable.getIntrinsicHeight():" + drawable.getIntrinsicHeight());
				imageView.setImageDrawable(drawable);
				Bitmap bitmap = drawable.getBitmap();
				FileOutputStream fos = null;
				try {
					fos = openFileOutput("image.png", Context.MODE_PRIVATE);
					bitmap.compress(Bitmap.CompressFormat.PNG, 100, fos);
				} catch (FileNotFoundException e) {
				} finally {
					if (fos != null) {
						try {
							fos.flush();
							fos.close();
						} catch (IOException e) {
						}
					}
				}

				break;
			default:
				break;
		}
	}
}

File: app/src/main/java/com/example/lvchen/myapplication/MainActivity.kt
package com.example.lvchen.myapplication

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.GravityCompat
import com.alibaba.android.arouter.launcher.ARouter
import com.example.lvchen.myapplication.ui.*
import com.example.lvchen.myapplication.ui.recyclerview.RecyclerViewRefreshActivity
import com.example.lvchen.myapplication.utils.KeyboardUtils
import com.example.lvchen.myapplication.view.TelNumCheckerView
import com.google.android.material.navigation.NavigationView
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.app_bar_main.*
import kotlinx.android.synthetic.main.content_main.*


/**
 * @author lvchen
 */
class MainActivity : AppCompatActivity(),
    NavigationView.OnNavigationItemSelectedListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)
        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                .setAction("Action", null)
                .show()
            val intent = Intent(Intent.ACTION_VIEW)
            val str =
                "suning://m.suning.com/index?adTypeCode=1002&adId=https%3A%2F%2Fcuxiao.m.suning.com%2Fscms%2Fcw03.html%3Futm_source%3Ddsp-ay%26utm_medium%3Das-cw1-30ll2-in5%26utm_campaign%3D%252C3%252Ca691f49313b24ea197107a1406aa9cf7%26utm_term%3DfZjq4KzfGgsu3k5D6IPl0PsfGaHe58%26adtype%3D7&utm_source=dsp-ay&utm_medium=as-cw1-30ll2-in5&utm_campaign=%2C3%2Ca691f49313b24ea197107a1406aa9cf7&utm_term=fZjq4KzfGgsu3k5D6IPl0PsfGaHe58&adtype=7"
            val str1 = "http://www.baidu.com"
            intent.data = Uri.parse(str)
            if (intent.resolveActivity(packageManager) != null) {
                startActivity(intent)
            } else {
                intent.data = Uri.parse(str1)
            }
            startActivity(intent)
        }

        val toggle = ActionBarDrawerToggle(
            this, drawer_layout, toolbar, R.string.navigation_drawer_open,
            R.string.navigation_drawer_close
        )
        drawer_layout.addDrawerListener(toggle)
        toggle.syncState()
        nav_view.setNavigationItemSelectedListener(this)

        viewpager_btn.setOnClickListener {
            startActivity(
                Intent(this, ViewPagerActivity::class.java)
            )
        }

        notification_btn.setOnClickListener {
            startActivity(
                Intent(this, NotificationActivity::class.java)
            )
        }

        wallpaper_btn.setOnClickListener {
            startActivity(
                Intent(this, WallPaperActivity::class.java)
            )
        }
        catalogue.setOnClickListener {
            startActivity(
                Intent(this, CatalogueActivity::class.java)
            )
        }
        catalogue2.setOnClickListener {
            startActivity(
                Intent(this, Catalogue2Activity::class.java)
            )
        }
        catalogue2.setOnClickListener {
            startActivity(
                Intent(this, Catalogue2Activity::class.java)
            )
        }
        waterfall.setOnClickListener {
            startActivity(
                Intent(this, WaterFallActivity::class.java)
            )
        }
        router_btn.setOnClickListener {
            ARouter.getInstance()
                .build("/test/activity2")
//          .withInt("audit", 666)
//          .withString("comment", "888")
                .navigation()
        }
        room_btn.setOnClickListener {
            startActivity(
                Intent(this, RoomActivity::class.java)
            )
        }

        cycle_head_btn.setOnClickListener {
            startActivity(
                Intent(this, CycleHeadActivity::class.java)
            )
        }
        chips.setOnClickListener {
            startActivity(
                Intent(this, ChipActivity::class.java)
            )
        }
        recycle_btn.setOnClickListener {
            startActivity(
                Intent(this, RecyclerViewRefreshActivity::class.java)
            )
        }
        verify_btn.setOnClickListener {

            val dialog = AlertDialog.Builder(this,R.style.inputDialog).apply {
                setView(TelNumCheckerView(context))
                setPositiveButton(
                    "确定"
                ) { dialog, which ->
                }
                setNegativeButton(
                    "关闭"
                ) { dialog, which ->
                }
            }
            dialog.create().show()
        }
//    NetworkService.apiService.sendVerifyCode2("18512527462")
//        .enqueue(object :
//            Callback<ResponseBody> {
//          override fun onFailure(
//            call: Call<ResponseBody>,
//            t: Throwable
//          ) {
//            Logger.d(t.message)
//          }
//
//          override fun onResponse(
//            call: Call<ResponseBody>,
//            response: Response<ResponseBody>
//          ) {
//            Logger.d(response.body().toString())
//          }
//
//        })
    }

    override fun onBackPressed() {
        val drawer =
            findViewById<View>(R.id.drawer_layout) as androidx.drawerlayout.widget.DrawerLayout
        if (drawer.isDrawerOpen(GravityCompat.START)) {
            drawer.closeDrawer(GravityCompat.START)
        } else {
            super.onBackPressed()
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)

    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        // Handle navigation view item clicks here.
        when (item.itemId) {
//      R.id.nav_camera -> startActivity(Intent(this, FragmentActivity::class.java))
            R.id.nav_manage -> startActivity(Intent(this, CoordinatorLayoutActivity::class.java))
            R.id.nav_gallery -> startActivity(Intent(this, RecycleViewActivity::class.java))
            R.id.nav_slideshow -> startActivity(Intent(this, GranzortViewActivity::class.java))
//      R.id.nav_share -> IdaddySdk.start()
            R.id.nav_send -> startActivity(Intent(this, XiaoAiTestActivity::class.java))
        }

        val drawer =
            findViewById<View>(R.id.drawer_layout) as androidx.drawerlayout.widget.DrawerLayout
        drawer.closeDrawer(GravityCompat.START)
        return true
    }

}
